tiledb/sm/array/array_operations.cc

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file   array_operations.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements I/O operations which support the Array class.
+ *
+ * The data- and I/O-oriented operations of class `Array` have been
+ * intentionally separated to support two _standalone_ object libraries.
+ * This design eliminates cyclic dependencies in the build, as each module has
+ * its own dependency chain.
+ * Note that the `array_operations` object library will not be implemented
+ * until _after_ the `Query` object libraries on which it relies.
```

Comment:
This comment is too much like copypasta from what's in `class Array`. Each should focus on what's in it, not on what's not.

In particular, this file has non-member functions that don't need access to member variables. Also, the dependency on `class Query` should be elucidated.